provider/gce/environ_network_test.go

```diff
@@ -1,77 +1,177 @@
+// Copyright 2017 Canonical Ltd.
 // Licensed under the AGPLv3, see LICENCE file for details.
 
 package gce_test
 
 import (
+	"github.com/juju/errors"
 	jc "github.com/juju/testing/checkers"
+	// "github.com/juju/version"
+	"google.golang.org/api/compute/v1"
 	gc "gopkg.in/check.v1"
 
+	// "github.com/juju/juju/constraints"
+	"github.com/juju/juju/environs"
+	// "github.com/juju/juju/environs/tags"
+	"github.com/juju/juju/instance"
+	"github.com/juju/juju/network"
 	"github.com/juju/juju/provider/gce"
+	"github.com/juju/juju/provider/gce/google"
 )
 
 type environNetSuite struct {
 	gce.BaseSuite
+	NetEnv environs.NetworkingEnviron
 }
 
 var _ = gc.Suite(&environNetSuite{})
 
-func (s *environNetSuite) TestGlobalFirewallName(c *gc.C) {
-	uuid := s.Config.UUID()
-	fwname := gce.GlobalFirewallName(s.Env)
-
-	c.Check(fwname, gc.Equals, "juju-"+uuid)
+func (s *environNetSuite) SetUpTest(c *gc.C) {
+	s.BaseSuite.SetUpTest(c)
+	netEnv, ok := environs.SupportsNetworking(s.Env)
+	c.Assert(ok, jc.IsTrue)
+	s.NetEnv = netEnv
 }
 
-func (s *environNetSuite) TestOpenPorts(c *gc.C) {
-	err := s.Env.OpenPorts(s.Rules)
+func (s *environNetSuite) cannedZones() {
+	s.FakeConn.Zones = []google.AvailabilityZone{
+		google.NewZone("a-zone", google.StatusUp, "", ""),
+		google.NewZone("b-zone", google.StatusUp, "", ""),
+	}
+}
 
-	c.Check(err, jc.ErrorIsNil)
+func (s environNetSuite) cannedSubnets() {
+	s.FakeConn.Subnets = []*compute.Subnetwork{{
+		Id:          1234,
+		IpCidrRange: "10.0.10.0/24",
+		Name:        "go-team",
+		Network:     "https://www.googleapis.com/compute/v1/projects/sonic-youth/global/networks/go-team",
+		Region:      "https://www.googleapis.com/compute/v1/projects/sonic-youth/regions/asia-east1",
+		SelfLink:    "https://www.googleapis.com/compute/v1/projects/sonic-youth/regions/asia-east1/subnetworks/go-team",
+	}, {
+		Id:          1235,
+		IpCidrRange: "10.0.20.0/24",
+		Name:        "shellac",
+		Network:     "https://www.googleapis.com/compute/v1/projects/sonic-youth/global/networks/shellac",
+		Region:      "https://www.googleapis.com/compute/v1/projects/sonic-youth/regions/asia-east1",
+		SelfLink:    "https://www.googleapis.com/compute/v1/projects/sonic-youth/regions/asia-east1/subnetworks/shellac",
+	}}
 }
 
-func (s *environNetSuite) TestOpenPortsAPI(c *gc.C) {
-	fwname := gce.GlobalFirewallName(s.Env)
-	err := s.Env.OpenPorts(s.Rules)
+func (s *environNetSuite) TestGettingAllSubnets(c *gc.C) {
+	s.cannedZones()
+	s.cannedSubnets()
+
+	subnets, err := s.NetEnv.Subnets(instance.UnknownId, nil)
 	c.Assert(err, jc.ErrorIsNil)
 
-	c.Check(s.FakeConn.Calls, gc.HasLen, 1)
-	c.Check(s.FakeConn.Calls[0].FuncName, gc.Equals, "OpenPorts")
-	c.Check(s.FakeConn.Calls[0].FirewallName, gc.Equals, fwname)
-	c.Check(s.FakeConn.Calls[0].Rules, jc.DeepEquals, s.Rules)
+	c.Assert(subnets, gc.DeepEquals, []network.SubnetInfo{{
+		ProviderId:        "https://www.googleapis.com/compute/v1/projects/sonic-youth/regions/asia-east1/subnetworks/go-team",
```

Comment:
Subnetworks have an integer ID which seems like it might be the right thing to use for Provider ID, but a network interface's Subnet field is the link rather than the ID (which I guess makes sense from a REST perspective). 
My thinking was that if I use the ID as the subnet ProviderId, then I won't be able to construct a `SubnetInfo` directly from the `NetworkInterface` in the instance version of Subnets - I'd need to do an extra call to get the subnets for an instance so I can get the ID.

But as I typed that it sounded wrong, and looking at the code I think it is - if I make the subnet ProviderId the integer ID everything should still work the same. I'll do that. I'd rather use the ID than trying to unpack components out of the URL.